source/Core/BSP/Miniware/configuration.h

```diff
@@ -63,8 +63,13 @@
  * OLED Brightness
  *
  */
-#define MIN_BRIGHTNESS     0   // Min OLED brightness selectable
-#define MAX_BRIGHTNESS     100 // Max OLED brightness selectable
+#if defined(MODEL_TS101)
+  #define MIN_BRIGHTNESS     1   // Min OLED brightness selectable
+  #define MAX_BRIGHTNESS     101 // Max OLED brightness selectable
```

Comment:
Hello. @Ralim & @discip, I have a technical question about this. The more I was thinking about this, the more I was getting very curious...

TL;DR: is it really safe & reliable to write `101` as brightness level directly to the _OLED hardware_?

- _OLED brightness_ is managed [here](https://github.com/Ralim/IronOS/blob/dev/source/Core/Src/settingsGUI.cpp#L872) by `displayBrightnessLevel()`
- then we jump into `OLED::setBrightness(...)` [here](https://github.com/Ralim/IronOS/blob/dev/source/Core/Drivers/OLED.cpp#L522)
- there we take a user "input" for brightness setting as [`contrast`](https://github.com/Ralim/IronOS/blob/dev/source/Core/Drivers/OLED.cpp#L524)
- and then, in short, as far as I understand, we fall through writing the setting value into _the OLED hardware_ through _software OLED driver_ using its register (which is allocated as a region of memory) using _I2C_ bus...

At the same time we have _"sanitizing"_ function to (re)set settings with validation which is [setSettingValue(...)](https://github.com/Ralim/IronOS/blob/dev/source/Core/Src/Settings.cpp#L179), where part of the settings matrix (min/max) is used. But! Setting _OLED brightness_ is not utilizing this "sanitization". Now, why it's important. _It depends on hardware_, so _some hardware_ _may be_ **fault tolerant** to wrong and/or fuzzing inputs but _**some may not**_. And something tells me that `100` as max brightness has been set originally based on datasheets of particular _OLED hardware_ used in the irons (haven't had time to dig so deep though yet).

Although I fully understand why minimum level for `TS101` should be `1`, is it OK to set maximum as `101` instead of original `100`?

As a safety measure, **one of two approaches** (_2a_ or _2b_) could be used for validation:

1. [set](https://github.com/ia/IronOSf/blob/6900fe81a181c02594bb1ce66928a07aed95db4c/source/Core/BSP/Miniware/configuration.h#L71) `MAX_BRIGHTNESS` back to `100` for _TS101_;
2. add validation (so even when it will be `1 + 25 + 25 + 25 + 25 = 101`, the actual setting for _the hardware_ still would be `100`):
  a. [directly](https://github.com/ia/IronOSf/blob/4da9015426743b2618d1991c1798bb2426d91f27/source/Core/Drivers/OLED.cpp#L523) inside of `OLED::setBrightness(...)`, i.e. `contrast = (MAX_BRIGHTNESS < contrast) ? MAX_BRIGHTNESS : contrast;`
  b. [indirectly](https://github.com/ia/IronOSf/blob/d1f60a7a966b691b76b11d969d971feb9c3b87a6/source/Core/Src/settingsGUI.cpp#L874-L876) inside of `displayBrightnessLevel()` using `setSettingValue()`, i.e. `setSettingValue(SettingsOptions::OLEDBrightness, value);` (the other calls of `OLED::setBrightness` either on start-up to set setting which is already there or via Bluetooth which is not related to _TS101_)

Should I make _pull-request_ with the check above or I shouldn't be worry much about this? Just really curious to hear any feedback on my thoughts (some parts describe very basic things but I did it only for my own better understanding).